packages/gossip/manager.go

```diff
@@ -359,5 +352,9 @@ func (m *Manager) processMessageRequest(data []byte, nbr *Neighbor) {
 	}
 
 	// send the loaded message directly to the neighbor
-	_, _ = nbr.Write(marshal(&pb.Message{Data: msgBytes}))
+	packet := &pb.Packet{Body: &pb.Packet_Message{Message: &pb.Message{Data: msgBytes}}}
+	if err := nbr.write(packet); err != nil {
+		nbr.log.Warnw("Failed to send requested message back to the neighbor", "err", err)
+		nbr.close()
```

Comment:
Upon error, the write function should have already called a `disconnect()`, so essentially this `close()` call just waits on the workgroup of the neighbor **for read events**; correct?

Reply:
Yes